server/controllers/user.js

```diff
@@ -91,6 +91,22 @@ module.exports = {
     }
   },
 
+  async updateBalance(req, res) {
+    try {
+      const userId = req.body.userId,
+        credits = req.body.credits;
+      const user = await User.findById(userId);
+      await User.findOneAndUpdate(
+        { _id: userId },
+        { balance: user.balance + credits }
+      );
+      const newUser = await User.findById(userId);
```

Comment:
no problem doing this but can also simplify using {new:true} on findByIdAndUpdate to get the most updated user.

Reply:
Thanks! will rewrite